pkg/k8s/client.go

```diff
@@ -120,6 +122,9 @@ func (c *Client) ListSecrets() []*models.Secret {
 
 // UpdateSecret updates the given secret
 func (c *Client) UpdateSecret(ctx context.Context, secret *models.Secret) error {
```

Comment:
The k8s package shouldn've have any knowledge about OSM domain objects (models.Secret). Instead, it should accept a corev1.Secret.

The compute/kube/client.go.Client object is responsible for doing the translation between the 2

Reply:
Let's fix Get/List secret here as well.